### PR TITLE
Add retry for retrieving repo info

### PR DESCRIPTION
**Pull request by starbops**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

Cannot upgrade to the same version (master build).

**Solution:**

Add a simple retry to cover this by requeue the upgrade resource.

**Related Issue:**

#3444

**Test plan:**

1. Deploy a cluster with master code including this PR
2. Upgrade using the same ISO image
3. Upgrade should succeed

### CODE REVIEW

File: pkg/controller/master/upgrade/upgrade_controller.go [hunk 1]
```diff
@@ -166,10 +172,34 @@ func (h *upgradeHandler) OnChanged(key string, upgrade *harvesterv1.Upgrade) (*h
 		toUpdate.Status.SingleNode = singleNode
 
 		repoInfo, err := repo.getInfo()
+		// retry repo.getInfo by requeueing
 		if err != nil {
-			setUpgradeCompletedCondition(toUpdate, StateFailed, corev1.ConditionFalse, err.Error(), "")
-			return h.upgradeClient.Update(toUpdate)
+			retryCount := 0
```
Comment (guangbochen) [inline pkg/controller/master/upgrade/upgrade_controller.go]: maybe we can use the [k8s retry utils](https://pkg.go.dev/k8s.io/client-go/util/retry) instead

Comment (starbops) [reply pkg/controller/master/upgrade/upgrade_controller.go]: Updated, please check again. Thanks!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]
LGTM, thanks!

**Comment by bk201** [comment]
@mergifyio backport v1.1

**Pull request merged**